examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,180 @@
 """
-==============================================================
-Plot Ridge coefficients as a function of the L2 regularization
-==============================================================
-
-.. currentmodule:: sklearn.linear_model
-
-:class:`Ridge` Regression is the estimator used in this example.
-Each color in the left plot represents one different dimension of the
-coefficient vector, and this is displayed as a function of the
-regularization parameter. The right plot shows how exact the solution
-is. This example illustrates how a well defined solution is
-found by Ridge regression and how regularization affects the
-coefficients and their values. The plot on the right shows how
-the difference of the coefficients from the estimator changes
-as a function of regularization.
-
-In this example the dependent variable Y is set as a function
-of the input features: y = X*w + c. The coefficient vector w is
-randomly sampled from a normal distribution, whereas the bias term c is
-set to a constant.
-
-As alpha tends toward zero the coefficients found by Ridge
-regression stabilize towards the randomly sampled vector w.
-For big alpha (strong regularisation) the coefficients
-are smaller (eventually converging at 0) leading to a
-simpler and biased solution.
-These dependencies can be observed on the left plot.
-
-The right plot shows the mean squared error between the
-coefficients found by the model and the chosen vector w.
-Less regularised models retrieve the exact
-coefficients (error is equal to 0), stronger regularised
-models increase the error.
-
-Please note that in this example the data is non-noisy, hence
-it is possible to extract the exact coefficients.
-
+=========================================================
+Ridge coefficients as a function of the L2 Regularization
+=========================================================
+
+A model that overfits learns the training data too well, capturing both the
+underlying patterns and the noise in the data. However, when applied to unseen
+data, the learned associations may not hold. We normally detect this when we
+apply our trained predictions to the test data and see the statistical
+performance drop significantly compared to the training data.
+
+One way to overcome overfitting is through regularization. Regularization
+penalizes large training weights, forcing the model to shrink all coefficients.
```

Comment:
```suggestion
One way to overcome overfitting is through regularization, which can be done by
penalizing large weights / coefficients in linear models, forcing the model to
shrink all coefficients.
```

WDYT @glemaitre ? Regularization in tree models is a very different thing for example.